test/test_computed_field.jl

```diff
@@ -301,7 +303,7 @@ function computations_with_averaged_field_derivative(model)
 
     set!(model, T = (x, y, z) -> 3 * z)
 
-    return all(interior(shear)[2:3, 2:3, 2:3] .== interior(T)[2:3, 2:3, 2:3])
+    return CUDA.@allowscalar all(interior(shear)[2:3, 2:3, 2:3] .== interior(T)[2:3, 2:3, 2:3])
```

Comment:
Same as above. We should minimize usage of `@allowscalar` in tests. This is one of the largest sources of technical debt in our tests and has incurred a lot of pain in the past